pkg/registry/events/event/strategy.go

```diff
@@ -56,8 +60,12 @@ func (eventStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Obje
 
 // Validate validates a new Event.
 func (eventStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
-	event := obj.(*api.Event)
-	return validation.ValidateEvent(event)
+	groupVersion := requestGroupVersion(ctx)
+	event := obj.(*coreapi.Event)
+	if groupVersion == eventsv1api.SchemeGroupVersion {
+		return eventsvalidation.ValidateEvent(event)
+	}
+	return corevalidation.ValidateEvent(event)
```

Comment:
So to clarify, do you mean I should remove the new added events/validation and instead, move additional checks for events/v1 into `ValidateEventCreate` under core/validation, so that we only need to maintain one validation file?

Reply:
Yes

Reply:
Done.